lib/internal.js

```diff
@@ -27,15 +27,20 @@ let clientObject;
         return this;
     }
 
-    function sendRequest(userOptions, optionalName,  requestType='single', responseType='single', validStatusCodes = [200, 201, 204, 404]){
+    function sendRequest(userOptions, operationCallback, optionalName, requestType='single', responseType='single'){
         if(!userOptions.path){
             throw new Error('Path is needed to send request.');
         }
         mlutil.copyProperties(userOptions,requestOptions);
         let operation = new Operation(
             optionalName?optionalName:'', clientObject, requestOptions, requestType, responseType
         );
-        operation.validStatusCodes = validStatusCodes;
+        if (operationCallback) {
+            operationCallback(operation);
+        }
+        if (!operation.validStatusCodes) {
```

Comment:
The thought is that if the user doesn't provide validStatusCodes, we'll default to these.